services/src/main/java/com/linkedin/kmf/consumer/KMBaseConsumer.java

```diff
@@ -14,6 +14,12 @@
  */
 public interface KMBaseConsumer {
 
+  /**
```

Comment:
OK.  The thread waiting for shutdown will not wait for the consumer thread to exit.  The consumer thread can be made a daemon thread.  This will allow consumers to block indefinitely.